engine/Shopware/Controllers/Backend/Article.php

```diff
@@ -157,7 +157,25 @@ public function saveAction()
     {
         $data = $this->Request()->getParams();
         if ($this->Request()->has('id')) {
+            /** @var Article $article */
             $article = $this->getRepository()->find((int) $this->Request()->getParam('id'));
+            // Check whether the article has been modified in the meantime
+            try {
+                $data['changed'] = new \DateTime($data['changed']);
+            } catch (\Exception $e) {
+                $data['changed'] = null;
+            }
+            if ($data['changed'] === null || $article->getChanged() != $data['changed']) {
```

Comment:
In which situation is `$data['changed'] === null`? Shouldn't we skip the check in this case?

Reply:
Ok I see, only on exception? We need to untangle the control flow a bit here...

Reply:
Suggestions?